lib/helpers.js

```diff
@@ -0,0 +1,121 @@
+'use strict';
+
+const globalsSet = require('./globals-set.js');
+let IMPORTS = {};
+
+/**
+ * Adds an import for the module to the list
+ *
+ * @param {string} mod - the name of the module to import from
+ * @param {string} original - the name of the original to get
+ */
+function updateImports(mod, original) {
+  if (mod in IMPORTS) {
+    if (original in IMPORTS[mod]) {
+      // if an import of original from module exists, noop
+      return;
+    }
+    // if an import from module exists without importing original,
+    // add original to the import
+    IMPORTS[mod].append(original);
+    return;
+  }
+
+  // if no import from module exists, add a new import
+  // of original, from module
+  IMPORTS[mod] = [original];
+}
+
+/**
+ * Adds all imports from the list to the top of the source file
+ *
+ * @param {SourceFile} sourceFile - the file to add the imports to
+ */
+function addAllImports(sourceFile) {
+  for (const module in IMPORTS) { // eslint-disable-line guard-for-in
+    sourceFile.addImportDeclaration({
+      defaultImport: IMPORTS[module][0],
+      moduleSpecifier: `std:global/${module}`,
+    });
+  }
+
+  IMPORTS = {};
+
+  /*
+    defaultImport?: string;
+    namespaceImport?: string;
+    namedImports?: (OptionalKind<ImportSpecifierStructure> |
+       string | WriterFunction)[] | WriterFunction;
+    moduleSpecifier: string;
+  */
+}
+
+/**
```

Comment:
Let's leave all these functions where they are, until we actually are using them in two source files. I suspect they will change when they do.

Then this file can just be something like `imports.js` or `imports-helpers.js` instead of a grab-bag full of all sorts of stuff.

Reply:
Hmm, I do favor changing the name to something more descriptive, but to me it doesn't really feel like a 'grab-bag', all these functions to me are critical for the core of handling the originals, checking, grabbing, and importing. I see the benefit of renaming, but I don't see the benefit of putting them back where they were.

Reply:
I think checking, grabbing, and importing are all very different. Putting them in one file is what makes it feel like a grab-bag.

In addition, all the non-imports functions right now are rather constructor-specific, so they should stay there until/unless they get generalized.